enterprise/api_client/enterprise_catalog.py

```diff
@@ -143,36 +143,68 @@ def delete_enterprise_catalog(self, catalog_uuid):
             return {}
 
     @JwtLmsApiClient.refresh_token
-    def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
+    def get_content_metadata(self, enterprise_customer=None, enterprise_catalogs=None, catalogs_last_modified=None):
         """
-        Return all content metadata contained in the catalogs associated with the EnterpriseCustomer.
+        Return all content metadata contained in the catalogs associated with the EnterpriseCustomer. Note: this method
+        handles two cases for content metadata retrieval. Although all parameters are listed as optional, a combination
+        is always required based on use case.
+
+        Use Case (1): If enterprise_customer is specified, then enterprise_catalogs is optional but must be a list of
+        EnterpriseCustomerCatalog objects if used. catalogs_last_modified cannot be used if enterprise_customer is
+        supplied. If enterprise_catalogs is used, this method will fetch content metadata for only the provided
+        catalogs, else all catalogs under the provided enterprise_customer will be fetched.
+
+        Use Case (2): If catalogs_last_modified is specified, then enterprise_catalogs MUST be provided. However,
+        enterprise_catalogs has be a list of dictionaries containing the a key/value pairs of `uuid: <catalog UUID>`.
```

Comment:
Would it be worth making a new function like `get_content_metadata_modified_since()` to make some of this branching logic easier to reason about?  It'd also help you get around the problem of sometimes returning one thing (a list) and sometimes two (a list and a timestamp).